test/support/adapter_assertions.ex

```diff
@@ -5,6 +5,24 @@ defmodule Cog.AdapterAssertions do
     assert response == expected_message
   end
 
+  # Because the raw template returns encoded json joined by newlines we have to
+  # add in newlines and then pull out the single result or return a list of
+  # result for those cases.
+  def assert_payload(%{"response" => response}, expected_payload) do
+    response = response
+    |> String.replace(~r/^}/m, "},")
+    |> String.rstrip(?,)
```

Comment:
Earlier, I wrote some code similar to this in the process of doing work that for a number of reasons ended up not being merged. I'd shared it with some other folks and thought it had been incorporated that way, but I can't seem to find it in the codebase.

```elixir
  # Currently, it's impossible to get a JSON array of data out of the
  # end of a pipeline. Instead, we get newline-separated JSON strings
  # of each individual data map in that array
  #
  # Example:
  #     "{\n  \"foo\": 123\n}\n{\n  \"foo\": 321\n}"
  #
  # This function recovers an Elixir data structure from this string,
  # by relying on the fact that the boundary between maps will be
  # "}\n{", which is done with a regex with positive lookahead and
  # lookbehind.
  #
  # Now you have `n` problems.
  defp unmangle_multiple_output(output_string),
    do: output_string |> String.split(~r/(?<=})\n(?={)/) |> Enum.map(&Poison.decode!(&1, keys: :atoms!))
```
I wonder if being stricter in the regex would be useful here.